pkg/destroy/ibmcloud/ibmcloud.go

```diff
@@ -0,0 +1,382 @@
+	"github.com/sirupsen/logrus"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/installer/pkg/destroy/providers"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/version"
+)
+
+var (
+	defaultTimeout = 2 * time.Minute
+)
+
+// ClusterUninstaller holds the various options for the cluster we want to delete
+type ClusterUninstaller struct {
+	ClusterName                      string
+	Context                          context.Context
+	Logger                           logrus.FieldLogger
+	InfraID                          string
+	AccountID                        string
+	BaseDomain                       string
+	CISInstanceCRN                   string
+	Region                           string
+	ResourceGroupName                string
+	UserProvidedSubnets              []string
+	UserProvidedVPC                  string
+	UserProvidedVPCResourceGroupName string
+
+	managementSvc          *resourcemanagerv2.ResourceManagerV2
+	controllerSvc          *resourcecontrollerv2.ResourceControllerV2
+	vpcSvc                 *vpcv1.VpcV1
+	iamPolicyManagementSvc *iampolicymanagementv1.IamPolicyManagementV1
+	zonesSvc               *zonesv1.ZonesV1
+	dnsRecordsSvc          *dnsrecordsv1.DnsRecordsV1
+
+	resourceGroupID string
+	cosInstanceID   string
+
+	errorTracker
+	pendingItemTracker
+}
+
+// New returns an IBMCloud destroyer from ClusterMetadata.
+func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.Destroyer, error) {
+	return &ClusterUninstaller{
+		ClusterName:                      metadata.ClusterName,
+		Context:                          context.Background(),
+		Logger:                           logger,
+		InfraID:                          metadata.InfraID,
+		AccountID:                        metadata.ClusterPlatformMetadata.IBMCloud.AccountID,
+		BaseDomain:                       metadata.ClusterPlatformMetadata.IBMCloud.BaseDomain,
+		CISInstanceCRN:                   metadata.ClusterPlatformMetadata.IBMCloud.CISInstanceCRN,
+		Region:                           metadata.ClusterPlatformMetadata.IBMCloud.Region,
+		ResourceGroupName:                metadata.ClusterPlatformMetadata.IBMCloud.ResourceGroupName,
+		UserProvidedSubnets:              metadata.ClusterPlatformMetadata.IBMCloud.Subnets,
+		UserProvidedVPC:                  metadata.ClusterPlatformMetadata.IBMCloud.VPC,
+		UserProvidedVPCResourceGroupName: metadata.ClusterPlatformMetadata.IBMCloud.ResourceGroupName,
+		pendingItemTracker:               newPendingItemTracker(),
+	}, nil
+}
+
+// Run is the entrypoint to start the uninstall process
+func (o *ClusterUninstaller) Run() error {
+	err := o.loadSDKServices()
+	if err != nil {
+		return err
+	}
+
+	err = o.destroyCluster()
+	if err != nil {
+		return errors.Wrap(err, "failed to destroy cluster")
+	}
+
+	return nil
+}
+
+func (o *ClusterUninstaller) destroyCluster() error {
+	stagedFuncs := [][]struct {
```

Comment:
Should this also delete the resource group, if created by the installer?

Reply:
Yeah. It should. Meant to hit that one as well.